tests/scale_test/utils.py

```diff
@@ -0,0 +1,238 @@
+    """
+    return str(uuid.uuid4())
+
+
+def harmonize_url(url):
+    """Harmonize the url string.
+
+    This function checks if url ends with slash, if not, add a slash to the url string.
+    Args:
+        url (str): A string of url.
+
+    Returns:
+        str: A string of url ends with slash.
+    """
+    url = str(url)
+    if not url.endswith('/'):
+        url += '/'
+    return url
+
+
+def is_https(url):
+    """HTTPS schema checker.
+
+    This function checks if the url string uses HTTPS schema.
+
+    Args:
+        url (str): A string of url.
+
+    Returns:
+        The return value. True for HTTPS, False otherwise.
+
+    """
+    if str(url).startswith('https://'):
+        return True
+    return False
+
+
+def read_bundles(bundle_list_file):
+    """Read bundles in to a dict from a file.
+
+    This function expects the JSON file follows a pre-defined schema:
+        [
+            {"bundle_uuid": XXX, "bundle_version": XXX},
+            {"bundle_uuid": XXX, "bundle_version": XXX}
+        ]
+
+    Args:
+        bundle_list_file (str): A path to the JSON file.
+
+    Returns:
+        dict: A dict of bundles listed in the file.
+    """
+    with open(bundle_list_file) as f:
+        bundles = json.load(f)
+    return bundles
+
+
+def required_checker(ctx, **kwargs):
+    """Check if all keyword arguments are provided by the user.
+
+    This function checks if all of the given keyword arguments are given by the user, if not, stop all of the current
+        commandline session. This custom checker function is better than the required=True flag of Click for this use
+        case since it also shows the full help text of the click commands.
+
+    Args:
+        ctx (click.Context): The special internal object that holds state relevant for the script execution at
+            every single level. This is automatically passed in by Click.
+        **kwargs: Arbitrary keyword command arguments.
+    """
+    for name, val in kwargs.items():
+        if not val:
+            click.echo('Error: Missing option "--{}".'.format(name))
+            click.echo(ctx.get_help())
+            ctx.exit()
+
+
+def auth_checker(ctx):
+    """A probe to check the auth_token.
+
+    This function performs as a probe, which validates the auth_token by sending a fake request to the Lira.
+
+    Args:
+        ctx (click.Context): The special internal object that holds state relevant for the script execution at
+            every single level. This is automatically passed in by Click.
+
+    Returns:
+        str: The ensured valid Lira auth_token.
+
+    """
+    auth_token = click.prompt('Please enter a valid auth_token for Lira (inputs are hidden)', type=str, hide_input=True)
+    probe_response = requests.post(ctx.obj['lira_url'] + 'notifications', params={'auth': auth_token}, json={'match': {}})
+    if probe_response.status_code == 401:
+        click.echo('Invalid auth token entered!')
+        ctx.exit()
+    else:
```

Comment:
What would happen if the notifications endpoint returns a non-401 error?

Reply:
I intentionally make it only check the 401 error code here. Because it will always return a 5xx error for the probe we sent from this script (the content of the auth-probe cannot be valid, or it will invoke a workflow). It could be better to check 404 here, but I don't think it is necessary since we always need to make sure the Lira we are using has the `notifications` endpoint. What do you think?

Reply:
Oh, I see! I think it's fine as it is then